synapse/storage/databases/main/cache.py

```diff
@@ -271,6 +297,106 @@ def _invalidate_caches_for_event(
             self._attempt_to_invalidate_cache("get_thread_participated", (relates_to,))
             self._attempt_to_invalidate_cache("get_threads", (room_id,))
 
+    def _invalidate_caches_for_events_and_stream(
```

Comment:
Maybe call this `_for_room_events_and_stream`? As written I'd guess the function accepts a collection of event IDs.